features/config.feature

```diff
@@ -92,6 +92,144 @@ Feature: Config
       The requested config file does not exist
       """
 
+  Scenario: PHP configuration file
+    Given a file named "behat.php" with:
+      """
+      <?php
+
+      use Behat\Config\Config;
+
+      return new Config();
+
+      """
+    And a file named "features/bootstrap/FeatureContext.php" with:
+      """
+      <?php
+
+      use Behat\Behat\Context\Context;
+
+      class FeatureContext implements Context
+      {
+      }
+      """
+    And a file named "features/config.feature" with:
+      """
+      Feature:
+        Scenario:
+          When this scenario executes
+      """
+    When I run "behat -f progress --no-colors --append-snippets"
+    Then it should pass with:
+      """
+      U
+
+      1 scenario (1 undefined)
+      1 step (1 undefined)
+
+      --- Use --snippets-for CLI option to generate snippets for following default suite steps:
+
+          When this scenario executes
+      """
+
+  Scenario: Alternative PHP configuration file
+    Given a file named "alternative-behat.php" with:
+      """
+      <?php
+
+      use Behat\Config\Config;
+
+      return new Config();
+
+      """
+    And a file named "features/bootstrap/FeatureContext.php" with:
+      """
+      <?php
+
+      use Behat\Behat\Context\Context;
+
+      class FeatureContext implements Context
+      {
+      }
+      """
+    And a file named "features/config.feature" with:
+      """
+      Feature:
+        Scenario:
+          When this scenario executes
+      """
+    When I run "behat -f progress --no-colors --append-snippets --config=alternative-behat.php"
+    Then it should pass with:
+      """
+      U
+
+      1 scenario (1 undefined)
+      1 step (1 undefined)
+
+      --- Use --snippets-for CLI option to generate snippets for following default suite steps:
+
+          When this scenario executes
+      """
+
+  Scenario: Custom PHP configuration object
+    Given a file named "custom-config-object.php" with:
+      """
+      <?php
+
+      use Behat\Config\ConfigInterface;
+
+      final class CustomConfig implements ConfigInterface
+      {
+        public function toArray(): array
+        {
+          return [
+            'default' => [
+              'gherkin' => [
+                'filters' => [
+                  'tags' => '@second'
+                ],
+              ],
+            ],
+          ];
```

Comment:
this is OK as-is, but it does make the example more verbose which slightly hides (for me) what's actually being tested. I would consider proving with a simpler config e.g. formatter (or `testers => [strict => true]]` if we want to use something different to the earlier examples).

Reply:
done